api/graphql/types/reservations/serializers/create_serializers.py

```diff
@@ -191,7 +192,10 @@ def validate(self, data, prefill_from_profile=True):
         reservation_unit_ids = [x.pk for x in reservation_units]
         self.check_reservation_type(user, reservation_unit_ids, reservation_type)
 
-        if settings.PREFILL_RESERVATION_WITH_PROFILE_DATA and prefill_from_profile:
+        prefill_from_profile = prefill_from_profile and settings.PREFILL_RESERVATION_WITH_PROFILE_DATA
+        id_token = get_id_token(user) or {}
+
+        if prefill_from_profile and is_ad_login(id_token):
```

Comment:
Negation is missing, should attempt the prefill only for non-AD users

Reply:
Had it the wrong way around. Fixed.